dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeer.cpp

```diff
@@ -337,9 +337,16 @@ FastAddPeerRes FastAddPeerImpl(
     UInt64 new_peer_id,
     UInt64 start_time)
 {
+    auto log = Logger::get("FastAddPeer");
     try
     {
-        auto elapsed = fap_ctx->tasks_trace->queryElapsed(region_id);
+        auto maybe_elapsed = fap_ctx->tasks_trace->queryElapsed(region_id);
+        if unlikely (!maybe_elapsed.has_value())
+        {
+            GET_METRIC(tiflash_fap_task_result, type_failed_cancel).Increment();
+            LOG_INFO(log, "FAP is canceled at beginning region_id={} new_peer_id={}", region_id, new_peer_id);
+        }
```

Comment:
In what situation will we run into this branch?

Reply:
When a cancel happens in `addTaskWithCancel` between `(*p)()` and `if (cancel_handle->canceled())`. It is very slightly chance.